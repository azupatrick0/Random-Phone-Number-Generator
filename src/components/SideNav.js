import React from 'react';
import Button from './Button';

const SideNav = ({ className, displaySideNav, sortPhoneNumbersAscending, sortPhoneNumbersDescending, generateCSV }) => (
	<div className={className}>
		<span
			style={{ color: 'red', fontSize: 40, display: className === 'side-nav-mobile' ? 'block' : 'none' }}
			onClick={displaySideNav}
		>
			&times;
		</span>
		<p
			style={{
				textAlign: 'center',
				fontWeight: 'bold',
				fontFamily: 'Verdana',
				color: 'white',
				fontSize: 20
			}}
		>
			Sort Phone Numbers
		</p>
		<div style={{ textAlign: 'center' }}>
			<Button
				btnBackgroundColor="#0000bb"
				btnText="ASCENDING ORDER"
				onClick={sortPhoneNumbersAscending}
				width="92%"
				border="0"
			/>
			<Button
				btnBackgroundColor="#aa0000"
				btnText="DESCENDING ORDER"
				onClick={sortPhoneNumbersDescending}
				width="92%"
				border="0"
			/>
			<p
				style={{
					textAlign: 'center',
					fontWeight: 'bold',
					fontFamily: 'Verdana',
					color: 'white',
					fontSize: 20
				}}
			>
				Export Phone Numbers
			</p>
			<Button
				btnBackgroundColor="#0000bb"
				btnText="EXPORT PHONE NUMBERS"
				onClick={generateCSV}
				width="92%"
				border="0"
			/>
		</div>
		<span className="side-nav__footer">
			Built with &#x2665; by{' '}
			<a
				href="https://www.linkedin.com/in/patrick-azu-22028a14b/"
				style={{ color: 'white', textDecoration: 'none' }}
			>
				Patrick Azu
			</a>
		</span>
	</div>
);

export default SideNav;
