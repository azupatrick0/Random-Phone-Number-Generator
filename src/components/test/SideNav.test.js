import React from 'react';
import SideNav from '../SideNav';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';

configure({ adapter: new Adapter() });

it('renders SideNav component without crashing', () => {
  const wrapper = shallow(<SideNav />);
  expect(wrapper).toMatchSnapshot();
});
