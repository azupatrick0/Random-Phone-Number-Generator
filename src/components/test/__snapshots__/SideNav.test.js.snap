// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SideNav component without crashing 1`] = `ShallowWrapper {}`;
